Source/Diagnostics/ReducedDiags/FieldProbe.cpp

```diff
@@ -147,102 +189,219 @@ void FieldProbe::ComputeDiags (int step)
+                ParticleReal* const AMREX_RESTRICT part_Ey = attribs[ParticleVal::Ey].dataPtr();
+                ParticleReal* const AMREX_RESTRICT part_Ez = attribs[ParticleVal::Ez].dataPtr();
+                ParticleReal* const AMREX_RESTRICT part_Bx = attribs[ParticleVal::Bx].dataPtr();
+                ParticleReal* const AMREX_RESTRICT part_By = attribs[ParticleVal::By].dataPtr();
+                ParticleReal* const AMREX_RESTRICT part_Bz = attribs[ParticleVal::Bz].dataPtr();
+                ParticleReal* const AMREX_RESTRICT part_S = attribs[ParticleVal::S].dataPtr();
 
                 amrex::Vector<amrex::Real> v_galilean{amrex::Vector<amrex::Real>(3, amrex::Real(0.))};
-                const auto &xyzmin = WarpX::GetInstance().LowerCornerWithGalilean(box, v_galilean,
-                                                                                  lev);
+                const auto &xyzmin = WarpX::GetInstance().LowerCornerWithGalilean(box, v_galilean, lev);
                 const std::array<Real, 3> &dx = WarpX::CellSize(lev);
+
                 const amrex::GpuArray<amrex::Real, 3> dx_arr = {dx[0], dx[1], dx[2]};
                 const amrex::GpuArray<amrex::Real, 3> xyzmin_arr = {xyzmin[0], xyzmin[1], xyzmin[2]};
 
-                //Interpolating to the probe point
-#if (AMREX_SPACEDIM == 2)
-                constexpr amrex::Real y_probe = 0._rt;
-#endif
-                doGatherShapeN(x_probe, y_probe, z_probe, Ex_interp, Ey_interp, Ez_interp,
-                               Bx_interp, By_interp, Bz_interp, arrEx, arrEy, arrEz, arrBx,
-                               arrBy, arrBz, Extype, Eytype, Eztype, Bxtype, Bytype, Bztype, dx_arr,
-                               xyzmin_arr, amrex::lbound(box), WarpX::n_rz_azimuthal_modes,
-                               interp_order, false);
-
-                // Either save the interpolated fields or the raw fields depending on the raw_fields flag
-                fp_values[index_Ex] = raw_fields ? arrEx(i_probe, j_probe, k_probe) : Ex_interp;
-                fp_values[index_Ey] = raw_fields ? arrEy(i_probe, j_probe, k_probe) : Ey_interp;
-                fp_values[index_Ez] = raw_fields ? arrEz(i_probe, j_probe, k_probe) : Ez_interp;
-                fp_values[index_Bx] = raw_fields ? arrBx(i_probe, j_probe, k_probe) : Bx_interp;
-                fp_values[index_By] = raw_fields ? arrBy(i_probe, j_probe, k_probe) : By_interp;
-                fp_values[index_Bz] = raw_fields ? arrBz(i_probe, j_probe, k_probe) : Bz_interp;
+                // Interpolating to the probe positions for each particle
+
+                // Temporarily defining modes and interp outside ParallelFor to avoid GPU compilation errors.
+                const int temp_modes = WarpX::n_rz_azimuthal_modes;
+                const int temp_interp_order = interp_order;
+                const bool temp_raw_fields = raw_fields;
+                const bool temp_field_probe_integrate = m_field_probe_integrate;
```

Comment:
Move up so we can use it in the kernel:
```suggestion
                const bool temp_field_probe_integrate = m_field_probe_integrate;
                amrex::Real const dt = WarpX::GetInstance().getdt(lev);
```